### PR TITLE
Updated TypeDoc plugins

### PR DESCRIPTION
**Pull request by tgreyuk**
## What is does?

This PR updates TypeDoc plugin versions to fix issues:

https://github.com/tgreyuk/typedoc-plugin-markdown/issues/475
https://github.com/tgreyuk/typedoc-plugin-markdown/issues/473
https://github.com/tgreyuk/typedoc-plugin-markdown/issues/470

## Changes

- Updates relevant typedoc plugin versions
- Updated plugin options
- Updated sidebar file to pull in manually created typedoc sidebar
- Updated referenced md files with new file names

### CODE REVIEW

**Comment by Montoya** [comment]
@ziad-saab nice changes here to get the sidebar working.

**Comment by ziad-saab** [comment]
Previewed locally, looks great! Thank you @tgreyuk for your plugins, and your contributions to our docs 🙏🏼❤️

**Review by ziad-saab — APPROVED** [review]

File: docusaurus.config.js [hunk 1]
```diff
@@ -86,11 +86,11 @@ const config = {
         entryPoints: ["./external/keyring-api/src/index.ts"],
         tsconfig: "./external/keyring-api/tsconfig.json",
         out: path.join(__dirname, "snaps/reference/keyring-api"),
-        sidebar: {
-          categoryLabel: "Keyring API",
-          position: 99,
-        },
-        identifiersAsCodeBlocks: true,
+        sidebar: { filteredIds: ["reference/keyring-api/index"] },
```
Comment (shanejonas) [inline docusaurus.config.js:89]: linting issue here. not sure why CI didnt catch it

Comment (ziad-saab) [reply docusaurus.config.js]: @shanejonas fixed.

**Review by shanejonas — APPROVED** [review]

**Comment by tgreyuk** [comment]
> Previewed locally, looks great! Thank you @tgreyuk for your plugins, and your contributions to our docs 🙏🏼❤️

No problem. Please do continue raising issues and suggested improvements - its very useful to have real feedback.

**Pull request merged**